### PR TITLE
Fixes #4 - Add Option to provide a different Piwik(tracker)-URL, e.g. if...

### LINKED ISSUE #4: Add Option to provide a different Piwik(tracker)-URL, e.g. if Piwik is behind a loadbalancer

### PR DESCRIPTION
**Pull request by peterbo**
... Piwik is behind a loadbalancer.

#4

### CODE REVIEW

File: Commands/GenerateVisits.php [hunk 1]
```diff
@@ -95,6 +98,19 @@ private function checkDays(InputInterface $input)
         return $days;
     }
 
+    private function checkCustomPiwikUrl(InputInterface $input)
+    {
+        $customPiwikUrl = $input->getOption('custom-piwik-url');
+
+		if(null === $customPiwikUrl) return null;
```
Comment (mattab) [inline Commands/GenerateVisits.php]: you can wrap this in curly brace and remove extra spaces

File: Controller.php [hunk 2]
```diff
@@ -107,6 +111,17 @@ private function checkDays()
         return $daysToCompute;
     }
 
+    private function checkCustomPiwikUrl()
+    {
+        $customPiwikUrl = Common::getRequestVar('customPiwikUrl', false, 'string');
+
+        if (!$customPiwikUrl = PageUrl::getUrlIfLookValid($customPiwikUrl)) {
```
Comment (mattab) [inline Controller.php]: the method `PageUrl::getUrlIfLookValid` is not tagged with `@api` - ideally plugins will only use public apis method. here and in the other `checkCustomPiwikUrl` you can use the @api method: `UrlHelper::isLookLikeUrl` (I just added `@api` to this method in core)

**Comment by mattab** [comment]
Hi @peterbo 

Feedback:
 * looks good to me
 * can you change your whitespace settings to be 4 spaces instead of tab?
 * wrote couple other small feedback

**Comment by peterbo** [comment]
Thanks for your feedback @mattab! I changed the code-style - I hope it's setup correctly now. I'll commit the changes tomorrow.

Have a nice day!

**Comment by mattab** [comment]
I've just realised when viewing the [screenshot test difference](http://builds-artifacts.piwik.org/ui-tests.master/11891.7/screenshot-diffs/singlediff.html?processed=../processed-ui-screenshots/UIIntegrationTest_admin_visitor_generator.png&expected=UIIntegrationTest_admin_visitor_generator.png&github=UIIntegrationTest_admin_visitor_generator.png) that maybe it's not 100% good idea to have the custom piwik URL in the UI as it makes it really trivial to "fake bomb" any piwik server... 

@peterbo maybe we need to remove the new field from the UI (but leave it in console command) to be safe...

**Comment by peterbo** [comment]
Hi Matt, this can be a risk indeed. I removed the input field from the UI (and from the controller) here: https://github.com/piwik/plugin-VisitorGenerator/pull/6

**Comment by mattab** [comment]
nice!

**Pull request merged**